src/dual-kad-dht.js

```diff
@@ -333,7 +334,12 @@ class DualKadDHT extends EventEmitter {
       throw errCode(new Error('Failed to load public key'), 'ERR_FAILED_TO_LOAD_KEY')
     }
 
-    const peerId = new PeerId(peer.id, undefined, pk)
+    const peerId = await PeerId.createFromPubKey(pk)
+
+    if (peerId.toB58String() !== peer.toB58String()) {
```

Comment:
We only create the PeerId from the public key to add it to the key book.  Do we even want to be doing that at this level?

Reply:
AFAIK it's only used when you're dealing with a peer with an RSA key (not Ed25519) and they should become less common over time?